source/ios/tools/IOSFeed/IOSFeed.cs

```diff
@@ -0,0 +1,146 @@
+            if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
+            {
+                // If the connection string is valid, proceed with operations against Blob
+                // storage here.
+                // Create the CloudBlobClient that represents the 
+                // Blob storage endpoint for the storage account.
+                CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+
+                CloudBlobContainer cloudBlobContainer =
+                    cloudBlobClient.GetContainerReference(Constants.ContainerId);
+                await cloudBlobContainer.CreateIfNotExistsAsync();
+
+                // Set the permissions so the blobs are public.
+                BlobContainerPermissions permissions = new BlobContainerPermissions
+                {
+                    PublicAccess = BlobContainerPublicAccessType.Blob
+                };
+
+                await cloudBlobContainer.SetPermissionsAsync(permissions);
+
+                var sourceFile = Path.Combine(Constants.FrameworkPath, Constants.FrameworkName);
+                var blobGuid = Guid.NewGuid().ToString();
+                var cloudFileName = blobGuid + Constants.FrameworkName;
+
+                CloudBlockBlob cloudBlockBlob = cloudBlobContainer.GetBlockBlobReference(cloudFileName);
+
+                await cloudBlockBlob.UploadFromFileAsync(sourceFile);
+
+                BlobContinuationToken blobContinuationToken = null;
+                do
+                {
+                    var results = await cloudBlobContainer.ListBlobsSegmentedAsync(null, blobContinuationToken);
+                    // Get the value of the continuation token returned by the listing call.
+                    blobContinuationToken = results.ContinuationToken;
+                    foreach (IListBlobItem item in results.Results)
+                    {
+                        var uriString = item.Uri.ToString();
+                        if (uriString.Contains(blobGuid))
+                        {
+                            UpdatePodSpec(uriString);
+                        }
+                    }
+                } while (blobContinuationToken != null);
+            }
+            else
+            {
+                throw new InvalidOperationException("invalid connection string");
+            }
+        }
+
+        private static void UpdatePodSpec(string uri)
+        {
+            var sourceFile = Path.Combine(Constants.PodspecPath, Constants.PodspecName);
+            var targetFile = Path.Combine(Constants.TargetPodspecPath, Constants.PodspecName);
+
+            FileInfo fi = new FileInfo(sourceFile);
+            if (!fi.Exists)
+            {
+                throw new FileNotFoundException("missing podspec");
+            }
+
+            // Open the stream and read it back.
+            using (StreamReader sr = File.OpenText(sourceFile))
+            {
+                string s = "";
```

Comment:
>string [](start = 16, length = 6)

strings in c# are immutable. That means every time you assign a new value to s a new string ojbect will be created under the covers; that will be grossly inefficient. Prefer StringBuilder here.

Reply:
thanks

---
In reply to: [314078732](https://github.com/microsoft/AdaptiveCards/pull/3291#discussion_r314078732) [](ancestors = 314078732)